### PR TITLE
LANG-1066: Add shell/XSI escape/unescape support

### PR DESCRIPTION
**Pull request by PascalSchumacher**
patch supplied by Mark

### CODE REVIEW

File: src/main/java/org/apache/commons/lang3/StringEscapeUtils.java [hunk 1]
```diff
@@ -272,6 +272,40 @@ public int translate(final CharSequence input, final int index, final Writer out
...
+              {"\\", "\\\\"},
+              {"\"", "\\\""},
+              {"'", "\\'"},
+              {" ", "\\ "},
+              {"\t", "\\\t"},
+              {"\r\n", ""},
+              {"\n", ""},
+              {"*", "\\*"},
+              {"?", "\\?"},
+              {"[", "\\["},
```
Comment (britter) [inline src/main/java/org/apache/commons/lang3/StringEscapeUtils.java]: What about the closing braked?

File: src/main/java/org/apache/commons/lang3/StringEscapeUtils.java [hunk 2]
```diff
@@ -413,6 +447,47 @@ public int translate(final CharSequence input, final int index, final Writer out
...
     }
 
+    public static final CharSequenceTranslator UNESCAPE_XSI = new BackslashUnescaper();
+
+    /**
+     * Translator object for unescaping backslash escaped entries.
+     *
+     * @since 3.6
+     */
+    static class BackslashUnescaper extends CharSequenceTranslator {
```
Comment (britter) [inline src/main/java/org/apache/commons/lang3/StringEscapeUtils.java]: Why not XsiUnescaper?

File: src/main/java/org/apache/commons/lang3/StringEscapeUtils.java [hunk 3]
```diff
@@ -801,4 +876,61 @@ public static final String unescapeCsv(final String input) {
...
+    }
+
+    /**
+     * <p>Alias for {@link #escapeXSI(String)}.</p>
+     *
+     * @param input String to escape values in, may be null
+     * @return String with escaped values, {@code null} if null string input
+     * @since 3.6
+     */
+    public static final String escapeShell(final String input) {
```
Comment (britter) [inline src/main/java/org/apache/commons/lang3/StringEscapeUtils.java]: I'm not sure whether it is a good idea to have aliases for methods. Does this help users to find what they are looking for or does this really cause confusion?

Comment (PascalSchumacher) [inline src/main/java/org/apache/commons/lang3/StringEscapeUtils.java]: Not sure. I'm fine with removing the methods.

File: src/main/java/org/apache/commons/lang3/StringEscapeUtils.java (hunk 2 above)
Comment (PascalSchumacher) [inline src/main/java/org/apache/commons/lang3/StringEscapeUtils.java]: No sure why the author of the patch choose this naming. I'm fine with changing it.

**Comment by PascalSchumacher** [comment]
As discussed on the mailing list string escaping will move to commons-text, so I'm closing this pull request.

**Pull request closed without merging**